src/server/zset_family.cc

```diff
@@ -2599,6 +2646,301 @@ void ZSetFamily::GeoDist(CmdArgList args, ConnectionContext* cntx) {
+
+  neighbors[0] = n.hash;
+  neighbors[1] = n.neighbors.north;
+  neighbors[2] = n.neighbors.south;
+  neighbors[3] = n.neighbors.east;
+  neighbors[4] = n.neighbors.west;
+  neighbors[5] = n.neighbors.north_east;
+  neighbors[6] = n.neighbors.north_west;
+  neighbors[7] = n.neighbors.south_east;
+  neighbors[8] = n.neighbors.south_west;
+
+  // Get range_specs for neighbors (*and* our own hashbox)
+  std::vector<ZSetFamily::ZRangeSpec> range_specs;
+  for (unsigned int i = 0; i < neighbors.size(); i++) {
+    if (HASHISZERO(neighbors[i])) {
+      continue;
+    }
+
+    // When a huge Radius (in the 5000 km range or more) is used,
+    // adjacent neighbors can be the same, leading to duplicated
+    // elements. Skip every range which is the same as the one
+    // processed previously.
+    if (last_processed && neighbors[i].bits == neighbors[last_processed].bits &&
+        neighbors[i].step == neighbors[last_processed].step) {
+      continue;
+    }
+
+    GeoHashFix52Bits min, max;
+    scoresOfGeoHashBox(neighbors[i], &min, &max);
+
+    ZSetFamily::ScoreInterval si;
+    si.first = ZSetFamily::Bound{static_cast<double>(min), false};
+    si.second = ZSetFamily::Bound{static_cast<double>(max), true};
+
+    ZSetFamily::RangeParams range_params;
+    range_params.interval_type = ZSetFamily::RangeParams::IntervalType::SCORE;
+    range_params.with_scores = true;
+    range_specs.emplace_back(si, range_params);
+
+    last_processed = i;
+  }
+
+  // get all the matching members and add them to the potential result list
+  auto cb = [&](Transaction* t, EngineShard* shard) {
+    return OpRanges(range_specs, t->GetOpArgs(shard), key);
+  };
+  OpResult<vector<ScoredArray>> result_arrays =
+      cntx->transaction->ScheduleSingleHopT(std::move(cb));
+  if (result_arrays.status() == OpStatus::WRONG_TYPE) {
+    (*cntx)->SendError(kWrongTypeErr);
+    return -1;
+  }
+
+  // filter potential result list
+  unsigned long count = 0;
+  double xy[2];
+  double distance;
+  for (auto& arr : *result_arrays) {
+    for (auto& p : arr) {
+      if (geoWithinShape(shape, p.second, xy, &distance) == 0) {
+        ga->emplace_back(xy[0], xy[1], distance, p.second, p.first);
+        count++;
+        if (limit > 0 && ga->size() >= limit)
+          break;
+      }
+    }
+  }
+
+  return count;
+}
+
+void GeoSearchGeneric(ConnectionContext* cntx, const GeoShape& shape_ref, string_view key,
+                      const GeoSearchOpts& geo_ops) {
+  double conversion = geo_ops.conversion;
+  uint64_t count = geo_ops.count;
+  bool asc = geo_ops.asc;
+  bool desc = geo_ops.desc;
+  bool any = geo_ops.any;
+  bool withdist = geo_ops.withdist;
+  bool withcoord = geo_ops.withcoord;
+  bool withhash = geo_ops.withhash;
+
+  // query
+  GeoShape* shape = &(const_cast<GeoShape&>(shape_ref));
+  GeoHashRadius georadius = geohashCalculateAreasByShapeWGS84(shape);
+  GeoArray ga;
+  MembersOfAllNeighbors(cntx, key, georadius, shape_ref, &ga, any ? count : 0);
```

Comment:
This function will reply to the client in case of error, so you have to check its return value here. We can't reply twice.
Further, it seems like you only case about the error case of the function, so no need to calculate and return `count` there

Reply:
I modified `MembersOfAllNeighbors` to return a bool value. It would return false if already replied inside.